src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.h

```diff
@@ -2,20 +2,108 @@
+#define ENSGORO_BOMBBUYFLAG_TALKED_HUMAN_FINALNIGHT (1 << 4)
+#define ENSGORO_BOMBBUYFLAG_TALKED_GOROKPK_FINALNIGHT (1 << 5)
+#define ENSGORO_BOMBBUYFLAG_TALKED_DEKU_FINALNIGHT (1 << 6)
+#define ENSGORO_BOMBBUYFLAG_TALKED_ZORA_FINALNIGHT (1 << 7)
+#define ENSGORO_BOMBBUYFLAG_TALKED_GORON (1 << 8)
+#define ENSGORO_BOMBBUYFLAG_TALKED_GORON_FINALNIGHT (1 << 9)
+#define ENSGORO_BOMBBUYFLAG_YESBUY (1 << 10)
 
 struct EnSGoro;
 
 typedef void (*EnSGoroActionFunc)(struct EnSGoro*, PlayState*);
 
+extern const ActorInit En_S_Goro_InitVars;
+
 typedef struct EnSGoro {
     /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x48];
+    /* 0x0144 */ Actor* otherGoron;
+    /* 0x0148 */ SkelAnime skelAnime;
     /* 0x018C */ EnSGoroActionFunc actionFunc;
-    /* 0x0190 */ char unk_190[0x180];
+    /* 0x0190 */ ColliderCylinder collider;
+    /* 0x01DC */ u16 actionFlags;
+    /* 0x01DE */ u16 bombbuyFlags;
+    /* 0x01E0 */ UNK_TYPE1 unk_1E0[0xC]; //No idea what this is.
+    /* 0x01EC */ s16 eyeTexIndex;
+    /* 0x01EE */ s16 eyeTimer;
+    /* 0x01F0 */ s16 frameTimer; //Counter?. Used in transformLimbDraw to determine first bool to pass to F298
+    /* 0x01F4 */ s32 loadedObjIndex;
+    /* 0x01F8 */ Vec3s jointTable[GORON_LIMB_MAX];
+    /* 0x0264 */ Vec3s morphTable[GORON_LIMB_MAX];
+    /* 0x02D0 */ f32 scaleFactor;
+    /* 0x02D4 */ s16 headRotZ; //transformLimbDraw, head
+    /* 0x02D6 */ s16 headRotY; //transformLimbDraw, head
+    /* 0x02D8 */ s16 bodyRotZ; //transformLimbDraw, head, body
+    /* 0x02DA */ s16 bodyRotY; //transformLimbDraw, body
+    /* 0x02DC */ Vec3f headTranslate; //transformLimbDraw, head
+    /* 0x02E8 */ Vec3s headRotate; //transformLimbDraw, head
+    /* 0x02F0 */ Vec3f bodyTranslate; //transformLimbDraw, body
+    /* 0x02FC */ Vec3s bodyRotate; //transformLimbDraw, body
+	/* 0x0302 */ UNK_TYPE2 unk_302; //Probably padding
+    /* 0x0304 */ u16 textID;
+    /* 0x0306 */ s16 snorePhase;
+    /* 0x0308 */ s32 animInfoIndex;
+    /* 0x030C */ s16 powderKegPrice;
 } EnSGoro; // size = 0x310
 
-extern const ActorInit En_S_Goro_InitVars;
+typedef enum EnSGoro_Type{
+	ENSGORO_TYPE_SHRINE_WINTER_A,
+	ENSGORO_TYPE_SHRINE_WINTER_B,
+	ENSGORO_TYPE_SHRINE_WINTER_C,
+	ENSGORO_TYPE_SHRINE_SPRING_A,
+	ENSGORO_TYPE_SHRINE_SPRING_B,
+	ENSGORO_TYPE_SHRINE_SPRING_C,
+	ENSGORO_TYPE_SHRINE_SPRING_D,
+	ENSGORO_TYPE_SHRINE_SPRING_E,
+	ENSGORO_TYPE_SHRINE_SPRING_F,
+	ENSGORO_TYPE_BOMBSHOP,
+} EnSGoro_Type;
+
+typedef enum EnSGoro_EyeTexture{
+	ENSGORO_EYETEX_OPEN,
+	ENSGORO_EYETEX_HALF,
+	ENSGORO_EYETEX_CLOSED,
+	ENSGORO_EYETEX_CLOSED2,
+} EnSGoro_EyeTexture;
+
+typedef enum EnSGoro_Animation{
+	ENSGORO_ANIM_IDLE_LIEDOWN_A,
+	ENSGORO_ANIM_IDLE_LIEDOWN_B,
+	ENSGORO_ANIM_UNROLL_A,
+	ENSGORO_ANIM_UNROLL_B,
+	ENSGORO_ANIM_ROLLUP,
+	ENSGORO_ANIM_SHIVER_A,
+	ENSGORO_ANIM_SHIVER_B,
+	ENSGORO_ANIM_COVEREARS = 11,
+	ENSGORO_ANIM_TAISOU_CHEER,
+	ENSGORO_ANIM_STAND_HANDTAP,
+	ENSGORO_ANIM_SLEEPY,
+	ENSGORO_ANIM_IDLE_STAND,
+} EnSGoro_Animation;
```

Comment:
Same here, enum val comments, PascalCase for the name, and these should go in `*.c` file since these two enums should only be used by the actor itself.